mars/deploy/oscar/session.py

```diff
@@ -1085,6 +1091,8 @@ async def fetch(self, *tileables, **kwargs) -> list:
             unexpected_keys = ", ".join(list(kwargs.keys()))
             raise TypeError(f"`fetch` got unexpected arguments: {unexpected_keys}")
 
+        fetcher = Fetcher.create(self._backend, get_storage_api=self._get_storage_api)
```

Comment:
Why pass this function instead of defining it in `Fetcher`?

Reply:
Because the Fetcher needs to construct the `StorageAPI` or `WebStorageAPI` to fetch the data from each bands. The construction requires some context, for exmaple, the address、session id、request rewriter, ...